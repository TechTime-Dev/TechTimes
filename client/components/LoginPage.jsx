import React, { useState } from "react";
import { useDispatch } from "react-redux";
import { useNavigate } from "react-router-dom";
import "flowbite";

const LoginPage = () => {
  const navigate = useNavigate();
  const [username, setUserName] = useState("");
  const [password, setPassword] = useState("");

  const handleSubmit = (event) => {
    event.preventDefault();
    fetch("http://localhost:3000/authenticate", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        username,
        password,
      }),
    })
      .then((response) => response.json())
      .then((result) => {
        // Check result and see if that's correct
      });
    setUserName("");
    setPassword("");
    navigate("/news");
  };
  return (
    <div className="flex flex-col justify-center items-center h-screen">
      <div className="w-full max-w-sm justify-center items-center mb-8 ml-14">
        <img
          src={require("../../public/logo-removebg.png").default}
          width="300"
          height="300"
        />
      </div>
      <div className="w-full items-center justify-center max-w-md p-4 bg-white border border-gray-200 rounded-lg shadow sm:p-6 md:p-8 dark:bg-gray-800 dark:border-gray-700 ">
        <form className="space-y-6" onSubmit={handleSubmit}>
          <h5 className="text-xl font-medium text-gray-900 dark:text-white text-center">
            Sign in to Tech Times
          </h5>
          <div>
            <label
              htmlFor="username"
              className="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
            >
              Username/Email
            </label>
            <input
              type="text"
              id="username"
              className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-600 dark:border-gray-500 dark:placeholder-gray-400 dark:text-white"
              required
              value={username}
              onChange={(event) => setUserName(event.target.value)}
            />
          </div>
          <div>
            <label
              htmlFor="password"
              className="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
            >
              Password
            </label>
            <input
              type="password"
              name="password"
              id="password"
              className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-600 dark:border-gray-500 dark:placeholder-gray-400 dark:text-white"
              required
              value={password}
              onChange={(event) => setPassword(event.target.value)}
            />
          </div>
          <div className="flex items-start">
            <div className="flex items-start">
              <div className="flex items-center h-5">
                <input
                  id="remember"
                  type="checkbox"
                  value=""
                  className="w-4 h-4 border border-gray-300 rounded bg-gray-50 focus:ring-3 focus:ring-blue-300 dark:bg-gray-700 dark:border-gray-600 dark:focus:ring-blue-600 dark:ring-offset-gray-800 dark:focus:ring-offset-gray-800"
                  required
                />
              </div>
              <label
                htmlFor="remember"
                className="ms-2 text-sm font-medium text-gray-900 dark:text-gray-300"
              >
                Remember me
              </label>
            </div>
            <a
              href="#"
              className="ms-auto text-sm text-blue-700 hover:underline dark:text-blue-500"
            >
              Forget Password?
            </a>
          </div>
          <button
            type="submit"
            className="w-full text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
          >
            Login
          </button>
          <div className="text-sm font-medium text-gray-500 dark:text-gray-300">
            Not registered?{" "}
            <a
              href="#"
              className="text-blue-700 hover:underline dark:text-blue-500"
            >
              Create account
            </a>
          </div>
        </form>
      </div>
    </div>
  );
};

export default LoginPage;
