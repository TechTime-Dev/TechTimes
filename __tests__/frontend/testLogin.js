import React from "react";
import { useDispatch } from "react-redux";
import { useNavigate } from "react-router-dom";
import LoginPage from "../../client/components/LoginPage.jsx";
import { render, fireEvent } from "@testing-library/react";

global.fetch = jest.fn().mockResolvedValue({
  json: jest.fn().mockResolvedValue({ message: "Login successful" }),
});

jest.mock("react-redux", () => ({
  useDispatch: jest.fn(),
}));

jest.mock("react-router-dom", () => ({
  useNavigate: jest.fn(),
}));

jest.spyOn(global, "fetch").mockResolvedValue({
  json: jest.fn().mockResolvedValue({ message: "Login successful" }),
});

describe("LoginPage", () => {
  let useDispatchMock;
  let useNavigateMock;

  beforeEach(() => {
    useDispatchMock = jest.fn();
    useDispatch.mockReturnValue(useDispatchMock);

    useNavigateMock = jest.fn();
    useNavigate.mockReturnValue(useNavigateMock);
  });

  it("should render the login form", () => {
    const { asFragment } = render(<LoginPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should update state on input change", () => {
    const { getByLabelText } = render(<LoginPage />);
    const usernameInput = getByLabelText("Username/Email");
    const passwordInput = getByLabelText("Password");

    fireEvent.change(usernameInput, { target: { value: "testuser" } });
    fireEvent.change(passwordInput, { target: { value: "testpassword" } });

    expect(usernameInput.value).toBe("testuser");
    expect(passwordInput.value).toBe("testpassword");
  });
});
