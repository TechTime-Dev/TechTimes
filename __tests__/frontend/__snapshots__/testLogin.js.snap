// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should render the login form 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center h-screen"
  >
    <div
      class="w-full max-w-sm justify-center items-center mb-8 ml-14"
    >
      <img
        height="300"
        width="300"
      />
    </div>
    <div
      class="w-full items-center justify-center max-w-md p-4 bg-white border border-gray-200 rounded-lg shadow sm:p-6 md:p-8 dark:bg-gray-800 dark:border-gray-700 "
    >
      <form
        class="space-y-6"
      >
        <h5
          class="text-xl font-medium text-gray-900 dark:text-white text-center"
        >
          Sign in to Tech Times
        </h5>
        <div>
          <label
            class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
            for="username"
          >
            Username/Email
          </label>
          <input
            class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-600 dark:border-gray-500 dark:placeholder-gray-400 dark:text-white"
            id="username"
            required=""
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
            for="password"
          >
            Password
          </label>
          <input
            class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-600 dark:border-gray-500 dark:placeholder-gray-400 dark:text-white"
            id="password"
            name="password"
            required=""
            type="password"
            value=""
          />
        </div>
        <div
          class="flex items-start"
        >
          <div
            class="flex items-start"
          >
            <div
              class="flex items-center h-5"
            >
              <input
                class="w-4 h-4 border border-gray-300 rounded bg-gray-50 focus:ring-3 focus:ring-blue-300 dark:bg-gray-700 dark:border-gray-600 dark:focus:ring-blue-600 dark:ring-offset-gray-800 dark:focus:ring-offset-gray-800"
                id="remember"
                required=""
                type="checkbox"
                value=""
              />
            </div>
            <label
              class="ms-2 text-sm font-medium text-gray-900 dark:text-gray-300"
              for="remember"
            >
              Remember me
            </label>
          </div>
          <a
            class="ms-auto text-sm text-blue-700 hover:underline dark:text-blue-500"
            href="#"
          >
            Forget Password?
          </a>
        </div>
        <button
          class="w-full text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
          type="submit"
        >
          Login
        </button>
        <div
          class="text-sm font-medium text-gray-500 dark:text-gray-300"
        >
          Not registered? 
          <a
            class="text-blue-700 hover:underline dark:text-blue-500"
            href="#"
          >
            Create account
          </a>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
